cpp/arcticdb/processing/operation_dispatch_binary.hpp

```diff
@@ -265,30 +282,50 @@ VariantData visit_binary_comparator(const VariantData& left, const VariantData&
         return transform_to_placeholder(result);
         },
         [&] ([[maybe_unused]] const std::shared_ptr<Value>& l, [[maybe_unused]] const std::shared_ptr<Value>& r) ->VariantData  {
-        util::raise_rte("Two value inputs not accepted to binary comparators");
+        user_input::raise<ErrorCode::E_INVALID_USER_ARGUMENT>("Two value inputs not accepted to binary comparators");
+        return EmptyResult{};
         },
         [](const auto &, const auto&) -> VariantData {
-        util::raise_rte("Bitset/ValueSet inputs not accepted to binary comparators");
+        user_input::raise<ErrorCode::E_INVALID_USER_ARGUMENT>("Bitset/ValueSet inputs not accepted to binary comparators");
+        return EmptyResult{};
     }
     }, left, right);
 }
 
+template <typename Func>
+inline std::string binary_operation_column_name(std::string_view left_column, Func&& func, std::string_view right_column) {
+    return fmt::format("({} {} {})", left_column, func, right_column);
+}
+
+template <typename Func>
+inline std::string binary_operation_with_types_to_string(std::string_view left, const TypeDescriptor& type_left, Func&& func,
+                                                         std::string_view right, const TypeDescriptor& type_right,
+                                                         bool arguments_reversed = false) {
+    if (arguments_reversed) {
+        return fmt::format("{} ({}) {} {} ({})", right, get_user_friendly_type_string(type_right), func, left, get_user_friendly_type_string(type_left));
+    }
+    return fmt::format("{} ({}) {} {} ({})", left, get_user_friendly_type_string(type_left), func, right, get_user_friendly_type_string(type_right));
+}
```

Comment:
Can you move these two functions to the top of the file? It is odd that they appear in amongst the main methods